platform/docs/docs/migration-guide.md

```diff
@@ -0,0 +1,492 @@
+without rebuild of the app for use cases such as reading from one PACS and
+writing to another.
+
+## LifeCycle Hooks
+
+OHIF v2 had `preRegistration` hook for extensions for initialization. In OHIF v3 you have
+even more control using `onModeEnter` and `onModeExit` hooks on the extensions and on the modes.
+
+## Extensions
+
+Since extensions in OHIF v2 were the main way of customizing the viewer, we will spend some time
+below to explain how you can migrate your extensions to OHIF v3.
+
+### Default Extension
+
+Lots of common functionalities in the platform/core has been moved inside
+the `@ohif/extension-default` extension. This extension is loaded by default
+in the viewer and it provides the following functionalities:
+
+- common datasources such as DICOMWeb, DICOMLocal, and DICOMJSON datasource.
+- default measurement panel and panel study browser
+- common toolbar button layouts
+- common hanging protocol configurations
+
+### Cornerstone Extension
+
+In OHIF v2, the Cornerstone extension provided modules like Cornerstone ViewportModule, ToolbarModule, and CommandsModule for controlling viewport actions.
+It relied on `react-cornerstone-viewport` for rendering viewports, `cornerstone-tools` for tools, and `cornerstone-core` for core functionalities.
+
+However, in OHIF v3, there have been significant changes. The rendering and tooling logic has been migrated to a new library called [`Cornerstone3D`](https://github.com/cornerstonejs/cornerstone3D-beta/). This means that all viewport rendering and tool functionalities are now handled by Cornerstone3D.
+
+Additionally, in OHIF v3, the native support for 3D functionalities previously provided by the `vtk` extension has been integrated into Cornerstone3D. As a result, there is no longer a need to use the vtk extension.
+
+To migrate from OHIF v2 to OHIF v3:
+
+- Update your code to use Cornerstone3D for viewport rendering and tools. If you don't have any custom tools or commands, you most likely won't need to make any changes.
+- If you have custom tools in `cornerstone-tools`, you will need to migrate them to Cornerstone3D. We have migrated various tools from `cornerstone-tools` to Cornerstone3D and you can
+use them as a reference (the main difference for tools are that now all toolData are stored in the world coordinate system).
+- Remove any dependencies on `react-cornerstone-viewport`, `cornerstone-tools`, and `cornerstone-core`.
+- Replace any usages of 3D functionalities from the vtk extension with the native 3D support provided by Cornerstone3D using its `VolumeViewport` component.
+
+By following these steps, you can leverage the improved rendering and tooling capabilities of Cornerstone3D and eliminate the need for the vtk extension in OHIF v3.
+
+### DICOM Segmentation & DICOM RT
+
+In OHIF v3, the equivalent extensions for RT and SEG exists with similar logic, but with various bug fixes and improvements.
```

Comment:
Point to those bug fixes and improvements here. People might use this as an energy boost to migrate.